src/context/data.test.ts

```diff
@@ -17,4 +18,21 @@ describe('data', () => {
       expect(result).toHaveProperty('body', `{"data":{"name":"msw"}}`)
     })
   })
+  describe('given composed with error', () => {
+    let result: ReturnType<typeof response>
+
+    beforeAll(() => {
+      result = response(
+        data({ name: 'msw' }),
```

Comment:
What won't work with shallow merging in `json` here is

```js
res(
  data({ foo: 'bar'}),
  data({ bar: 'bar'})
)
```

as it arrives at `json` as `data` which is one level nested meaning here we would only get `{ bar: 'bar }` has that's how `json` merges. Just for awareness.

Reply:
That is incorrect behavior. We should merge _deeply_, with the latest `data` call taking priority. 

```js
res(data({ a: 1 }), data({ b: 2, c: { d: 3 } }))
// { data: { a: 1, b: 2, c: { d: 3 } } }
```